src/ncp/ncp_base.cpp

```diff
@@ -3494,13 +3569,40 @@ ThreadError NcpBase::SetPropertyHandler_MAC_SCAN_STATE(uint8_t header, spinel_pr
             break;
 
         case SPINEL_SCAN_STATE_ENERGY:
-            errorCode = otEnergyScan(
-                mInstance,
-                mChannelMask,
-                mScanPeriod,
-                &HandleEnergyScanResult_Jump,
-                this
-            );
+#if OPENTHREAD_ENABLE_RAW_LINK_API
+            if (otLinkRawIsEnabled(mInstance))
+            {
+                // Make sure we aren't already scanning and that we have
+                // only 1 bit set for the channel mask.
+                if (mCurScanChannel == NCP_INVALID_SCAN_CHANNEL &&
+                    HasOnly1BitSet(mChannelMask))
```

Comment:
The check of `mCurScanChannel` seems superfluous, since we would only get to this code if we weren't already in a scan. (If that isn't the case, then we should likely fix that)

`kThreadError_InvalidState` seems like an inappropriate error to return if there is more than one bit set. We should likely indicate some sort of invalid argument error.